### PR TITLE
Add Body::chain to create a Body from a series of two Body instances

### PR DESCRIPTION
**Pull request by joshtriplett**
This method helps when prepending or appending some data to a Body.

### CODE REVIEW

File: src/body.rs [hunk 1]
```diff
@@ -482,6 +482,46 @@ impl Body {
...
+    /// ```
+    #[cfg(feature = "async-std")]
+    pub fn chain(self, other: Body) -> Self {
+        let mime = if self.mime == other.mime {
+            self.mime.clone()
+        } else {
+            mime::BYTE_STREAM
+        };
+        let length = match (self.length, other.length) {
+            (Some(l1), Some(l2)) => (l1 - self.bytes_read).checked_add(l2 - other.bytes_read),
```
Comment (jbr) [inline src/body.rs:515]: bytes_read should always be zero at this point, and if not, that may indicate a larger issue. the body isn't read until the response is handed back to async-h1, at which point we do not expect mutation. additionally, if we've read from both bodies already, the two bodies will be interleaved and be corrupted. We should probably panic if either has a nonzero bytes read, as that's an invalid and irrecoverable state

Comment (joshtriplett) [reply src/body.rs:515]: @jbr I originally started with that approach, but I *think* this actually works. I can add some more tests to cover this case.

The specific use case that prompted me to not just assert at the top of `chain`: start a process, wrap its Reader in a Body, read some data from the front of that Reader, then prepend some other data in front of it. I *think* that'll work fine here; both `Body` instances will handle returning their remaining bytes.

Comment (joshtriplett) [reply src/body.rs:515]: I just added a test to make sure this case works correctly.

**Review by yoshuawuyts — APPROVED** [review]
neat!

**Pull request merged**